lib/DDG/Meta/Data.pm

```diff
@@ -36,91 +34,74 @@ unless(%ia_metadata){
-            }
+        # generic IsAwesome goodie metadata since these are always the same
+        if($ia->{perl_module} =~ /IsAwesome/){
+            next IA if $ia_metadata{module}{'DDG::Goodie::IsAwesome'};
+            $ia->{id} = 'is_awesome';
+            $ia->{perl_module} = 'DDG::Goodie::IsAwesome'
+        }
 
-            # warn if we run into a duplicate id.  These should be unique within *and*
-            # across all IA types
-            if( $ia_metadata{id}{$id} ){
-                warn "Duplicate ID for IA with ID: $id";
-            }
+        # warn if we run into a duplicate id.  These should be unique within *and*
+        # across all IA types
+        if( $ia_metadata{id}{$id} ){
+            warn "Duplicate ID for IA with ID: $id";
+        }
 
-            my $perl_module = $ia->{perl_module};
-
-            # Clean up/set some values
-            $ia->{signal_from} ||= $ia->{id};
-            $ia->{js_callback_name} = _js_callback_name($perl_module);
-
-            #add new ia to ia_metadata{id}
-            $ia_metadata{id}{$id} = $ia;
-            #add new ia to ia_metadata{module}. Multiple ias per module possible
-            push @{$ia_metadata{module}{$perl_module}}, $ia;
-            # by language for multilang wiki
-            if($is_fathead){
-                my $source = $ia->{src_id};
-                # by source number for fatheads
-                $ia_metadata{fathead_source}{$source} = $ia;
-                # by language for multi language wiki sources
-                # check that language is set since most fatheads don't have a language
-                if( my $lang = $ia->{src_options}{language}){
-                    $ia_metadata{fathead_lang}{$lang} = $source;
-                }
-                my $min_length = $ia->{src_options}{min_abstract_length};
-                $ia_metadata{fathead_min_length}{$source} = $min_length if $min_length;
+        my $perl_module = $ia->{perl_module};
+
+        # Clean up/set some values
+        $ia->{signal_from} ||= $ia->{id};
+        $ia->{js_callback_name} = _js_callback_name($perl_module);
+
+        #add new ia to ia_metadata{id}
+        $ia_metadata{id}{$id} = $ia;
+        #add new ia to ia_metadata{module}. Multiple ias per module possible
+        push @{$ia_metadata{module}{$perl_module}}, $ia;
+        # by language for multilang wiki
+        if($ia->{repo} eq 'fathead'){
+            my $source = $ia->{src_id};
+            # by source number for fatheads
+            $ia_metadata{fathead_source}{$source} = $ia;
+            # by language for multi language wiki sources
+            # check that language is set since most fatheads don't have a language
+            if( my $lang = $ia->{src_options}{language}){
+                $ia_metadata{fathead_lang}{$lang} = $source;
             }
+            my $min_length = $ia->{src_options}{min_abstract_length};
+            $ia_metadata{fathead_min_length}{$source} = $min_length if $min_length;
```

Comment:
@russellholt you can add `?w=1` to skip that BS: https://github.com/duckduckgo/duckduckgo/pull/150/files?w=1.